sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/BulkWriterITest.scala

```diff
@@ -410,6 +427,325 @@ class BulkWriterITest extends IntegrationSpec with CosmosClient with AutoCleanab
+    val writeConfig = CosmosWriteConfig(
+      ItemWriteStrategy.ItemOverwrite,
+      5,
+      bulkEnabled = true,
+      bulkMaxPendingOperations = Some(900)
+    )
+
+    val bulkWriter = new BulkWriter(container, partitionKeyDefinition, writeConfig, DiagnosticsConfig(Option.empty, false, None))
+
+    // First create one item, as patch can only operate on existing items
+    val itemWithFullSchema = CosmosPatchTestHelper.getPatchItemWithFullSchema(UUID.randomUUID().toString)
+    val id = itemWithFullSchema.get("id").textValue()
+    val partitionKey = new PartitionKey(id)
+
+    bulkWriter.scheduleWrite(partitionKey, itemWithFullSchema)
+    bulkWriter.flushAndClose()
+    // make sure the item exists
+    val originalItem = container.readItem(id, partitionKey, classOf[ObjectNode]).block().getItem()
+
+    // Patch operation will fail as it is trying to apply increment type for non-numeric type
+    try {
+      val incrementPartialUpdateInvalidSchema = StructType(Seq(
+        StructField("propInt", IntegerType),
+        StructField("propString", StringType),
+      ))
+
+      val columnConfigsMap = new TrieMap[String, CosmosPatchColumnConfig]
+      incrementPartialUpdateInvalidSchema.fields.foreach(field => {
+        columnConfigsMap += field.name -> CosmosPatchColumnConfig(field.name, CosmosPatchOperationTypes.Increment, s"/${field.name}")
+      })
+
+      val bulkWriterForPatch = CosmosPatchTestHelper.getBulkWriterForPatch(columnConfigsMap, container, partitionKeyDefinition)
+      val patchPartialUpdateItem = CosmosPatchTestHelper.getPatchItemWithSchema(id, incrementPartialUpdateInvalidSchema, originalItem)
+
+      bulkWriterForPatch.scheduleWrite(partitionKey, patchPartialUpdateItem)
+      bulkWriterForPatch.flushAndClose()
+    } catch {
+      case e: Exception => e.getMessage should startWith("Increment operation is not supported for type StringType")
+    }
+
+    // Patch operation will succeed as it only apply increment on numeric type
+    val incrementPartialUpdateValidSchema = StructType(Seq(
+      StructField("propInt", IntegerType),
+      StructField("propLong", LongType),
+      StructField("propFloat", FloatType),
+      StructField("propDouble", DoubleType)
+    ))
+
+    val columnConfigsMap = new TrieMap[String, CosmosPatchColumnConfig]
+    incrementPartialUpdateValidSchema.fields.foreach(field => {
+      columnConfigsMap += field.name -> CosmosPatchColumnConfig(field.name, CosmosPatchOperationTypes.Increment, s"/${field.name}")
+    })
+
+    val bulkWriterForPatch = CosmosPatchTestHelper.getBulkWriterForPatch(columnConfigsMap, container, partitionKeyDefinition)
+    val patchPartialUpdateItem = CosmosPatchTestHelper.getPatchItemWithSchema(id, incrementPartialUpdateValidSchema, originalItem)
+
+    bulkWriterForPatch.scheduleWrite(partitionKey, patchPartialUpdateItem)
+    bulkWriterForPatch.flushAndClose()
+  }
+
+  "Bulk Writer" can "skip partial update for cosmos system properties" in {
```

Comment:
Probably "should" be better here, but somehow after I use "should", the test after will not be executed, will figure out why